### PR TITLE
Add `filterNodeArgumentsForWorkerThreads` option

### LINKED ISSUE #3207: Add option to filter invalid arguments from `execArgv` when launching worker threads
Please provide details about:

* What you're trying to do

I'm using a task runner that's constructing Node commands as such:

```
node --title svc-test:test ../../node_modules/ava/entrypoints/cli.mjs
```

* What happened

`--title` is not valid for creating a worker. 

```
✘ Internal error
  Error [ERR_WORKER_INVALID_EXEC_ARGV]: Initiated Worker with invalid execArgv flags: --title
  Error [ERR_WORKER_INVALID_EXEC_ARGV]: Initiated Worker with invalid execArgv flags: --title
      at new NodeError (node:internal/errors:399:5)
      at new Worker (node:internal/worker:192:13)
      at createWorker (file:///Users/user/code/test/node_modules/.pnpm/ava@5.3.0/node_modules/ava/lib/fork.js:24:12)
      at loadFork (file:///Users/user/code/test/node_modules/.pnpm/ava@5.3.0/node_modules/ava/lib/fork.js:84:39)
      at pMap.concurrency.concurrency (file:///Users/user/code/test/node_modules/.pnpm/ava@5.3.0/node_modules/ava/lib/api.js:289:20)
      at file:///Users/user/code/test/node_modules/.pnpm/p-map@5.5.0/node_modules/p-map/index.js:141:26
```

* What you expected to happen

I'd expect that Ava would filter out options that would throw. Probably related https://github.com/nodejs/node/issues/41103#issuecomment-1054384840

### PR DESCRIPTION
**Pull request by bojavou**
Adds a configuration option `threadArgumentsFilter` for filtering `nodeArguments` sent to worker threads. The option is supported only in config files due to accepting a function. When threads are disabled, the filter is ignored and the full unfiltered list of arguments is sent to worker processes.

`main` is currently failing tests. This patch adds no new failures. All new tests pass. The new feature is documented in `06-configuration.md`.

This enabled me to use natives syntax in threads.

Closes #3207.

### CODE REVIEW

**Comment by bojavou** [comment]
This check is failing:

[Install and test AVA / Install dependencies without using a lockfile (pull_request)](https://github.com/avajs/ava/actions/runs/10050073126/job/27777287067?pr=3336)

I'm not clear why, and the new tests passed in the log. Is it possible this failure is unrelated?

File: docs/06-configuration.md [hunk 1]
```diff
@@ -337,3 +337,18 @@ These may also export a function which is then invoked, and can receive argument
 The `nodeArguments` configuration may be used to specify additional arguments for launching worker processes. These are combined with `--node-arguments` passed on the CLI and any arguments passed to the `node` binary when starting AVA.
 
 [CLI]: ./05-command-line.md
+
+## Thread arguments filter
+
+In a config file only, `threadArgumentsFilter` may provide a function used for filtering `nodeArguments` sent to worker threads. This enables excluding arguments that throw if sent to a thread. The filter is ignored by worker processes.
+
+```js
```
Comment (tommy-mitchell) [inline docs/06-configuration.md:345]:
```suggestion
`ava.config.js`:
```js
```

File: lib/cli.js [hunk 2]
```diff
@@ -388,9 +388,16 @@ export default async function loadCli() { // eslint-disable-line complexity
 		exit(error.message);
 	}
 
+	const workerThreads = combined.workerThreads !== false;
+
 	let nodeArguments;
 	try {
 		nodeArguments = normalizeNodeArguments(conf.nodeArguments, argv['node-arguments']);
+		nodeArguments ??= process.execArgv;
```
Comment (novemberborn) [inline lib/cli.js:396]: `normalizeNodeArguments` always returns an array, which starts with the `execArgv` values.

```suggestion
```

File: docs/06-configuration.md [hunk 3]
```diff
@@ -337,3 +337,19 @@ These may also export a function which is then invoked, and can receive argument
...
+
+`ava.config.js`:
+```js
+const processOnly = new Set([
+	'--allow-natives-syntax',
+	'--expose-gc'
+]);
+
+export default {
+	threadArgumentsFilter: argument => !processOnly.has(argument)
```
Comment (novemberborn) [inline docs/06-configuration.md:353]: With apologies for the bike-shedding, is `filterNodeArgumentsForWorkerThreads` a little clearer, albeit rather verbose?

**Review by novemberborn — CHANGES_REQUESTED** [review]
Nice! Don't worry about unrelated test breakage, I can resolve those.

File: lib/cli.js (hunk 2 above)
Comment (bojavou) [reply lib/cli.js:396]: Can't believe I missed that.

File: docs/06-configuration.md (hunk 3 above)
Comment (bojavou) [reply docs/06-configuration.md:353]: I'm just rerunning tests with this change.

**Comment by novemberborn** [comment]
The Linux tests compare reporter output against a known-good version. That specific suite only runs on Linux because it's finicky enough as it is, without adding other environments into the mix.

However, it's failing through what looks like a change in this PR:

```
  [31m✘ Internal error[39m
  TypeError: execArgv is not iterable

  [90mTypeError: execArgv is not iterable[39m
  [90m    at createWorker (/lib/fork.js:49:18)[39m
  [90m    at loadFork (/lib/fork.js:79:39)[39m
  [90m    at pMap.concurrency.concurrency (/lib/api.js:285:20)[39m
  [90m    at /node_modules/p-map/index.js:109:26[39m
```

**Comment by bojavou** [comment]
Not really sure what to change here but I'm moving toward an internal test runner anyway so I'm just dropping it.

**Comment by novemberborn** [comment]
That's fine, I can take over.

**Pull request merged**